pkg/checks/dns/metrics.go

```diff
@@ -0,0 +1,86 @@
+// sparrow
+// (C) 2024, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dns
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// metrics defines the metric collectors of the DNS check
+type metrics struct {
+	status    *prometheus.GaugeVec
+	duration  *prometheus.GaugeVec
+	count     *prometheus.CounterVec
+	histogram *prometheus.HistogramVec
+}
+
+// newMetrics initializes metric collectors of the dns check
+func newMetrics() metrics {
+	return metrics{
+		status: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "sparrow_dns_status",
+				Help: "Specifies if the target can be resolved.",
+			},
+			[]string{"target"},
+		),
+		duration: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "sparrow_dns_duration_seconds",
```

Comment:
Shouldn't the duration and histogram metrics have the same name, since they measure the same thing? Prometheus does add the _count, _bucket and _sum to the histrogram metrics exposed, so it shouldn't be a problem.

Reply:
We should test it, I have in mind that the go prom collector is panicking if the name of a metric is a duplicate.